### ISSUE TITLE
Optimal memory size for the script

### ISSUE THREAD

**Issue by seakha** [digitalmethodsinitiative/dmi-tcat#136]
In your experience what is the optimal memory size for the script. In the config.php file 2GB is defined as the max memory limit for the script. Have you ever faced a situation that the script might need more memory?

**Comment by ErikBorra** [comment]
Hi @seakha,

it really depends on the usage of DMI-TCAT. Memory is mainly used for buffering MySQL queries. If you have a lot of simultaneous users you'll want more memory assigned to MySQL. If you have big data sets, you'll want more memory assigned to MySQL. Most important is the configuration of memory in MySQL. As stated on the installation instructions, "We recommend using the script available on http://mysqltuner.com to tweak your mysql config, and to set both the sort_buffer_size and myisam_sort_buffer_size as big as possible."

That said, I just noticed that the [automatic installation script](https://raw.githubusercontent.com/digitalmethodsinitiative/dmi-tcat/master/helpers/tcat-deploy-debian.sh) does not properly set my.conf yet. Assigning this to @dentoir .

As for the memory assigned to PHP; it does not need to be that high (2G). Most data sets are not read fully into memory anyway. We just set it high enough to start with so that we're quite sure that users don't run into problems (assuming they have more than 2G of RAM).

Best,

Erik
